evm/src/all_stark.rs

```diff
@@ -85,31 +87,40 @@ mod tests {
         let keccak_stark = KeccakStark::<F, D> {
             f: Default::default(),
         };
-        let keccak_rows = 256;
+        let keccak_rows = (NUM_ROUNDS + 1).next_power_of_two();
         let keccak_looked_col = 3;
 
+        let mut rng = ChaCha8Rng::seed_from_u64(0x6feb51b7ec230f25);
+        let num_inputs = 1;
+        let keccak_inputs = (0..num_inputs)
+            .map(|_| [0u64; INPUT_LIMBS].map(|_| rng.gen()))
+            .collect_vec();
+        let keccak_trace = keccak_stark.generate_trace(keccak_inputs);
+        let column_to_copy: Vec<_> = keccak_trace[keccak_looked_col].values[..].into();
+
+        let default = vec![F::ONE; 1];
+
         let mut cpu_trace_rows = vec![];
         for i in 0..cpu_rows {
             let mut cpu_trace_row = [F::ZERO; CpuStark::<F, D>::COLUMNS];
             cpu_trace_row[cpu::columns::IS_CPU_CYCLE] = F::ONE;
-            cpu_trace_row[cpu::columns::OPCODE] = F::from_canonical_usize(i);
+            if i < keccak_rows {
+                cpu_trace_row[cpu::columns::OPCODE] = column_to_copy[i];
+            } else {
+                cpu_trace_row[cpu::columns::OPCODE] = default[0];
+            }
             cpu_stark.generate(&mut cpu_trace_row);
             cpu_trace_rows.push(cpu_trace_row);
         }
         let cpu_trace = trace_rows_to_poly_values(cpu_trace_rows);
 
-        let mut keccak_trace =
-            vec![PolynomialValues::zero(keccak_rows); KeccakStark::<F, D>::COLUMNS];
-        keccak_trace[keccak_looked_col] = cpu_trace[cpu::columns::OPCODE].clone();
-
-        let default = vec![F::ZERO; 2];
-        let cross_table_lookups = vec![CrossTableLookup {
-            looking_tables: vec![Table::Cpu],
-            looking_columns: vec![vec![cpu::columns::OPCODE]],
-            looked_table: Table::Keccak,
-            looked_columns: vec![keccak_looked_col],
+        let cross_table_lookups = vec![CrossTableLookup::new(
+            vec![Table::Cpu],
+            vec![vec![cpu::columns::OPCODE]],
```

Comment:
Let's just add a TODO that this is a temporary thing until CTL filters are working

Reply:
Ah what exactly is temporary? The use of `default`?

Reply:
I mean that the `OPCODE` register won't be part of the actual CTL right?

Reply:
Ah yes